### PR TITLE
Changing message_modal behavior, fixing gene search submit

### PR DESCRIPTION
**Pull request by bistline**
The default behavior of the main "message_modal" has changed to always render into the page, whether there is any content to display or not.  Showing/hiding the modal has been abstracted out into the `showMessageModal()` function.  This fixes the issue where removing the open modal from the DOM to show a new message wedges the UI, and cannot be recovered.

Also, fixing a regression with submitting a gene search by pressing enter, rather than clicking the "Search Genes" button.

This PR satisfies [SCP-1272](https://broadworkbench.atlassian.net/browse/SCP-1272).

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -544,6 +544,43 @@ function closeModalSpinner(spinnerTarget, modalTarget, callback) {
     $(modalTarget).modal('hide');
 }
 
+// handles showing/hiding main message_modal and cleaning up state on full & partial page renders
+function showMessageModal(notice=null, alert=null) {
+    console.log('calling showMessageModal');
```
Comment (eweitz) [inline app/assets/javascripts/application.js]: [`console.log` in production](https://www.google.com/search?q=console.log+in+production) is generally discouraged.  It's fine in development, but in production omitting such debugging code is good form.  Breakpoints in DevTools are a cleaner and more robust solution if we need to debug in production.

```suggestion
```

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -544,6 +544,43 @@ function closeModalSpinner(spinnerTarget, modalTarget, callback) {
...
+        $("#message_modal").modal("show");
+    }
+
+    // don't timeout alert messages
+    if (alert) {
+        setTimeout(function() {
+            $("#message_modal").modal("hide");
+        }, 3000);
+    }
+}
```
Comment (eweitz) [inline app/assets/javascripts/application.js]: Does this code mean "if there is an alert, hide the message modal after 3 seconds"?  If so:

```suggestion
    // timeout alert messages
    if (alert) {
        setTimeout(function() {
            $("#message_modal").modal("hide");
        }, 3000);
    }
}
```

Elsewhere, I see `if (<%= alert.nil? %>) {` with another copy of this code snippet, which hints that maybe `if (!alert) {` is intended here.

Either way, either the code or the comment seems incorrect.

**Comment by eweitz** [comment]
Please add `security-risk` GitHub label.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (bistline) [reply app/assets/javascripts/application.js]: Yeah, that's a debug statement I forgot to take out.  Will remove.

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (bistline) [reply app/assets/javascripts/application.js]: You're correct - if there is an alert message present, it should not call `setTimeout()`.  I'll update that as well.

File: app/controllers/api/v1/search_controller.rb [hunk 3]
```diff
@@ -176,14 +178,19 @@ def index
...
         when :keyword
-          @studies = @studies.to_a.sort_by {|study| -study.search_weight(@search_terms.split) }
+          @studies = @studies.sort_by {|study| -study.search_weight(@search_terms.split) }
         when :accession
-          @studies = @studies.to_a.sort_by {|study| possible_accessions.index(study.accession) }
+          @studies = @studies.sort_by {|study| possible_accessions.index(study.accession) }
         when :facet
-          @studies = @studies.to_a.sort_by {|study| -@studies_by_facet[study.accession][:facet_search_weight]}
+          @studies = @studies.sort_by {|study| -@studies_by_facet[study.accession][:facet_search_weight]}
+        else
```
Comment (devonbush) [inline app/controllers/api/v1/search_controller.rb]: nice, I like what you've done here better a lot better than the way I had it

**Pull request merged**